content/articles/bigdecimal-and-primitive-data-type-arithmetic-program-in-java/index.md

```diff
@@ -0,0 +1,212 @@
+author: bonface-ndolo
+date: 2021-09-22T00:00:00-03:40
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/bigdecimal-and-primitive-data-type-arithmetic-program-in-java/hero.jpg
+    alt: BigDecimal and Primitive Data Type Arithmetic Program in Java Hero Image
+---
+
+Because of their inaccuracy, floating-point data types cannot be used in financial calculations. This is why Java provides a separate class named **BigDecimal** for executing operations and reducing the chances of calculation errors. 
+<!--more-->
+On double numbers, the BigDecimal class provides arithmetic, scale management, rounding, comparison, format conversion, and hashing functions. It compensates for the time complexity, however, by handling extremely large and extremely small floating-point integers with exceptional precision.
+
+These basic arithmetic operations can be performed on BigDecimal as well as between BigDecimal and primitive data types, which is what our article will cover. While both `BigInteger` and `BigDecimal` support arbitrary-precision integers, BigDecimal only supports arbitrary-precision fixed-point numbers. 
+
+BigDecimal/BigInteger is not for you if you're constructing a low-latency application where every microsecond counts. Let us look at our first example of Adding, subtracting, and multiplying two large decimal integers with a Java program:
+
+```Java
+import java.io.*;
+import java.math.BigDecimal;
+class Example {
+    public static void main(String[] args)
+    {
+
+        // Declaration of the bigdecimal numbers
+        BigDecimal first1 = new BigDecimal("25.35557");
+        BigDecimal sec2 = new BigDecimal("55.3767");
+         //actual arithmetic addition operation
+        System.out.println("Adding first1 and sec2 = "
+                           + (first1.add(sec2)));
+
+        // actual arithmetic subtraction operation
+        System.out.println("Subtracting first1 and sec2 = "
+                           + (first1.subtract(sec2)));
+
+        // actual arithmetic multiplication operation
+        System.out.println(
+            "Multiplying first1 and sec2 = "
+            + (first1.multiply(sec2)));
+    }
+}
+```
+
+Output:
+
+```
+Adding first1 and sec2 = 80.73227
+Subtracting first1 and sec2 = -30.02113
+Multiplying first1 and sec2 = 1404.107793
+```
+
+We have done three operations on our two declared BigDecimals. For example the subtraction, addition, and the addition operations. There is now a point to note here, we did not perform the division operation because of one reason that I will be giving.
+
+Because the division of the two numbers is non-terminating, there is an error that will occur, and we know that BigDecimal was established to provide the highest level of precision. As a result, it generates an error. 

```

Comment:
```suggestion
Because the division of the two numbers is non-terminating, there is an error that will occur. And, we know that BigDecimal was established to provide the highest level of precision. As a result, it generates an error.
```